### ISSUE TITLE
[Calendar] Switch months and type from parent component

### ISSUE THREAD

**Issue by zomars** [ant-design/ant-design#12992]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
It would be nice to be able to be able to let us handle the month and view switching in the Calendar component. This would allow us to keep multiple components in sync with a specific date or view type.

A more specific use case:

I'm building an events component that shares a split view between a List and a Calendar, the current month is selected from another component and I would need both the List and Calendar to display the month selected in my local state.

### What does the proposed API look like?
I'm thinking something like this:

```js
<Calendar
  currentMonth={this.state.currentMonth}
/> 
<Button onClick={() => this.setState({ currentMonth: any })}>Change month</Button>
```

**Comment by ztplz** [comment]
Currently Calendar receives a `value` that controls the date  displayed.  I think you can handle the month by changing `value`.

**Comment by zomars** [comment]
Thanks for your reply. Will try that. 

El 8 nov 2018, a la(s) 20:39, 偏右 <notifications@github.com> escribió:

Closed #12992.
